modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/render.es.js

```diff
@@ -62,7 +62,12 @@ export default function render(renderable, renderData, container) {
 			}
 		);
 
-		const Component = typeof renderable === 'function' ? renderable : null;
+		const Component =
+			typeof renderable === 'function'
+				? renderable
+				: typeof renderable.render === 'function'
+				? renderable.render
```

Comment:
I didn't check why... but either because of how we export or how we unbox in the loader, our `.default` for `ClayButton` (and I assume other clay components) is not **a** renderable right now but an object... 🤷

Reply:
Out of curiosity, what does the object look like?

Reply:
(If you can be bothered.)

Reply:
### render.es.js
<img width="489" alt="Screen Shot 2020-06-17 at 09 44 18" src="https://user-images.githubusercontent.com/905006/84870204-4a2cb300-b07f-11ea-8241-1748cbf32a6d.png">

### Button.js
<img width="392" alt="Screen Shot 2020-06-17 at 09 43 04" src="https://user-images.githubusercontent.com/905006/84870160-37b27980-b07f-11ea-8a4b-a0d905672e41.png">

Reply:
Sigh, so it's the `forwardRef` "HOC". We could do a check like this:

```js
renderable.$$typeof == Symbol.for('react.forward_ref')
```

but I am not sure if that is considered "internal" or not. The React dev tools rely on these names being stable, [it would seem](https://github.com/facebook/react/blob/9025949d840c89619bed9bf06f5c14c25c141c5a/packages/react-devtools-shared/src/backend/ReactSymbols.js#L35), so it is probably safe.

I wonder if there are any other common HOCs like this that we need to watch out for.

Reply:
Fixed at [Considers Forwardref HOC as a renderableConsiders Forwardref HOC as a renderable](https://github.com/liferay-frontend/liferay-portal/pull/2/commits/4a2488523b476313aac7e6f600b86342eae7cf47)